### PR TITLE
Expose SubstructLibrary serialization stream

### PR DESCRIPTION
**Pull request by bp-kelley**
Exposes the substruct library's streaming interface.  This is a bit maddening as you need to write to a text stream and read from a binary stream!

This is partly due to the python_streambuf.h implementation, but I've yet to find a work around.

```
      with open(path, 'w') as file:
        slib.ToStream(file)

      file = open(path, 'rb')
      slib2 = rdSubstructLibrary.SubstructLibrary()
      slib2.InitFromStream(file)
      self.assertEqual(len(slib), len(slib2))
```

If this seems Ok, I'll add better docs.

### CODE REVIEW

File: Code/GraphMol/SubstructLibrary/Wrap/SubstructLibraryWrap.cpp [hunk 1]
```diff
@@ -196,6 +198,18 @@ struct substructlibrary_pickle_suite : python::pickle_suite {
...
 };
 
+void toStream(const SubstructLibrary &cat, python::object &fileobj) {
+  streambuf ss(fileobj, 't');
+  streambuf::ostream ost(ss);
+  cat.toStream(ost);
+}
+
+  void initFromStream(SubstructLibrary &cat, python::object &fileobj) {
+  streambuf ss(fileobj);
```
Comment (greglandrum) [inline Code/GraphMol/SubstructLibrary/Wrap/SubstructLibraryWrap.cpp]: If you require a binary stream to read from, and it sounds like you do, then it's a good idea to include a `'b'` mode here. The `python::streambuf` code will check that the stream actually is opened in binary mode

Comment (bp-kelley) [reply Code/GraphMol/SubstructLibrary/Wrap/SubstructLibraryWrap.cpp]: It would also be good if the test was actually run :)

**Comment by bp-kelley** [comment]
I'm going to try to get the out stream to use binaryio as well for symmetry.

**Comment by greglandrum** [comment]
Ok, so you want me to hold off on merging?

**Comment by bp-kelley** [comment]
Yes, I need to add some docs anyway, this write to text and read from binary needs some explaining, it’s pretty ugly, def a python 3 wart IMHO.

----
Brian Kelley

**Comment by greglandrum** [comment]
Though I agree with you that it's not nice, it's the same story with the SDWriter and ForwardSDMolSupplier. We have informative error messages in place, so it's not super critical.

**Comment by bp-kelley** [comment]
@greglandrum this is ready.

**Comment by greglandrum** [comment]
@bp-kelley : looks like the CI failures are real problems with the python tests

**Comment by bp-kelley** [comment]
Looks like doc test issues, I’ll fix it.

**Pull request merged**